doc/en/user/source/datadirectory/migrating.rst

```diff
@@ -153,4 +153,33 @@ Backup files
                config.xml.2.2.x
           config.xml.2.2.x        
 
-       
\ No newline at end of file
+Migrating between GeoServer 2.5.x and 2.6.x
+-------------------------------------------
+
+The catalog naming conventions became more strict in 2.6, invalidating certain characters within names. This is because certain protocols will not work correctly with certain characters in the name. The naming restrictions are automatically set to relaxed through the STRICT_PATH java system property variable. In order to ensure your names will work with all protocols, set this variable to true. ::
```

Comment:
This is a bit hard to follow.  It seems like you mean that the default is relaxed and it can be overridden with the property.  Then you say "This will allow, with warnings".  It's ambiguous whether you mean the relaxed default allows or those characters, or that enabling strict mode enables the warnings.